subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -84,16 +84,18 @@ int zsock_getaddrinfo(const char *host, const char *service,
 		      struct zsock_addrinfo **res)
 {
 	int family = AF_UNSPEC;
-	long int port;
+	long int port = 0;
 	int st1 = DNS_EAI_ADDRFAMILY, st2 = DNS_EAI_ADDRFAMILY;
 
 	if (hints) {
 		family = hints->ai_family;
 	}
 
-	port = strtol(service, NULL, 10);
-	if (port < 1 || port > 65535) {
-		return DNS_EAI_NONAME;
+	if (service) {
+		port = strtol(service, NULL, 10);
+		if (port < 1 || port > 65535) {
```

Comment:
You can simplify this by writing this if as: `if ((long int)(u16_t)port != port)`.

Reply:
I did not intend to change the logic, just to check for a valid service pointer.

Changing to you suggestion will not catch when port is 0.  Checking ```errno``` should not be needed when checking a range without 0. The implementation of ```strtol()``` used here does not set ```errno``` when returning 0 because of no conversion.